libsolidity/interface/CompilerStack.cpp

```diff
@@ -243,26 +243,34 @@ bool CompilerStack::parse()
 	vector<string> sourcesToParse;
 	for (auto const& s: m_sources)
 		sourcesToParse.push_back(s.first);
-	for (size_t i = 0; i < sourcesToParse.size(); ++i)
+
+	try
 	{
-		string const& path = sourcesToParse[i];
-		Source& source = m_sources[path];
-		source.scanner->reset();
-		source.ast = parser.parse(source.scanner);
-		if (!source.ast)
-			solAssert(!Error::containsOnlyWarnings(m_errorReporter.errors()), "Parser returned null but did not report error.");
-		else
+		for (size_t i = 0; i < sourcesToParse.size(); ++i)
 		{
-			source.ast->annotation().path = path;
-			for (auto const& newSource: loadMissingSources(*source.ast, path))
+			string const& path = sourcesToParse[i];
+			Source& source = m_sources[path];
+			source.scanner->reset();
+			source.ast = parser.parse(source.scanner);
+			if (!source.ast)
+				solAssert(!Error::containsOnlyWarnings(m_errorReporter.errors()), "Parser returned null but did not report error.");
+			else
 			{
-				string const& newPath = newSource.first;
-				string const& newContents = newSource.second;
-				m_sources[newPath].scanner = make_shared<Scanner>(CharStream(newContents, newPath));
-				sourcesToParse.push_back(newPath);
+				source.ast->annotation().path = path;
+				for (auto const& newSource: loadMissingSources(*source.ast, path))
+				{
+					string const& newPath = newSource.first;
+					string const& newContents = newSource.second;
+					m_sources[newPath].scanner = make_shared<Scanner>(CharStream(newContents, newPath));
+					sourcesToParse.push_back(newPath);
+				}
 			}
 		}
 	}
+	catch(FatalError const&)
+	{
+		// no-op
```

Comment:
The yunciton `Parser::parse` correctly catches fatal errors, but `loadMissingSources` does not. Maybe it would be more consistent to catch the fatal error there.

Reply:
thats true..